### PR TITLE
Add abyss package

### PR DESCRIPTION
**Pull request by akthoma**

### CODE REVIEW

File: var/spack/repos/builtin/packages/abyss/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    homepage = "http://www.bcgsc.ca/platform/bioinfo/software/abyss"
+    url      = "http://www.bcgsc.ca/platform/bioinfo/software/abyss/releases/2.0.2/abyss-2.0.2.tar.gz"
+
+    version('2.0.2', '1623f55ad7f4586e80f6e74b1f27c798')
+
+    depends_on('openmpi', type='build')
+    depends_on('boost', type='build')
+    depends_on('sparsehash', type='build')
+    depends_on('sqlite', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/abyss/package.py]: I don't think `build` is the right dependency type for these dependencies. See http://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types for the meaning of these types. I would guess the default (build/link) is what you want.

File: var/spack/repos/builtin/packages/abyss/package.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    conflicts('boost', when='@1.51.0:1.52.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/abyss/package.py]: Your syntax is wrong here. What you're trying to do would look like:
```python
depends_on('boost@:1.50,1.53:')
```

File: var/spack/repos/builtin/packages/abyss/package.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    """ABySS is a de novo, parallel, paired-end sequence assembler
+       that is designed for short reads. The single-processor version
+       is useful for assembling genomes up to 100 Mbases in size."""
+
+    homepage = "http://www.bcgsc.ca/platform/bioinfo/software/abyss"
+    url      = "http://www.bcgsc.ca/platform/bioinfo/software/abyss/releases/2.0.2/abyss-2.0.2.tar.gz"
+
+    version('2.0.2', '1623f55ad7f4586e80f6e74b1f27c798')
+
+    depends_on('openmpi', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/abyss/package.py]: I wonder if other MPI providers will work for this package. The developers only mention OpenMPI: https://github.com/bcgsc/abyss#dependencies

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/abyss/package.py]: Oh, if you check the output of `configure --help`, you'll see that it does support MPICH (in Spack) and LAM (not in Spack) as well:
```
$ ./configure --help
...
  --enable-mpich          use MPICH (default is to use Open MPI)
  --enable-lammpi         use LAM/MPI (default is to use Open MPI)
```
You might want to change this to:
```python
depends_on('mpi')
```
and conflicts for the other MPI providers, and enable this flag `if 'mpich' in self.spec`.

Comment (akthoma) [inline var/spack/repos/builtin/packages/abyss/package.py]: How do I go about adding conflicts with every MPI provider except for openmpi and mpich?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/abyss/package.py]: @akthoma The syntax would look like:
```python
conflicts('^intel-mpi')
conflicts('^intel-parallel-studio+mpi')
conflicts('^mvapich2')
conflicts('^spectrum-mpi')
```
You can get a complete list of MPI providers by running:
```
$ spack providers mpi
```

Comment (akthoma) [inline var/spack/repos/builtin/packages/abyss/package.py]: Thank you very much!

**Pull request merged**